### DISCUSSION TITLE
curl Zigbee command JSON results and jq

### DISCUSSION THREAD

**Discussion by baszied** [arendst/Tasmota#18577] [Q&A] [answered]
Hi,

I flashed my Sonoff Zigbee Pro and created a number of scripts that now interface with this little box through http using curl. I use these scripts in Domoticsz and everything works just fine. It however came to my notice that the output of some zigbee commands is not correctly interpreted by jq. This makes me think that the actual returned JSON string is not correct...  Is it just me or am I making a mistake here ?

The command I send is simply:
`curl -s POST -u admin:$ZIGBEEPWD -d "cmnd=ZbInfo Sonoff-Mini-Z07" "http://$ZIGBEESVR/cm" | jq .`

This returns
```
{
  "ZbInfo": "Done"
}
```

The actual output without jq however is:
`{"ZbInfo":{"0xB117":{"Device":"0xB117","Name":"Sonoff-Mini-Z07","IEEEAddr":"0xDC8E95FFFE0D56B7","ModelId":"01MINIZB","Manufacturer":"SONOFF","Endpoints":[1,242],"Config":["O01"],"Power":0,"Reachable":false,"LastSeen":774577,"LastSeenEpoch":1682414775,"LinkQuality":58}},"ZbInfo":"Done"}`

So jq only returns 2nd result...
Perhaps its a bug of jq :)

**Comment by barbudor** [comment]
Tasmota is expecting user and password as arguments in the url, not as standard http authentication
See exemple in the doc
https://tasmota.github.io/docs/Commands/#with-web-requests

**Reply by barbudor** [reply]
Also it shouldn't be a post but a get with the command in the url
Please stick to the doc

**Reply by sfromis** [reply]
Also, it is recommended to use MQTT for automations, simpler, faster and less risk of trouble, like what you run into using `jq`. That tool may not be up to gracefully handling the "unusual" thing about the same key occurring more than once. Graceful handling could include rejecting the payload, instead of silently discarding data.

**Reply by sfromis** [reply, marked as answer]
On an operational level, the different JSON after setting the option `ZbNoPrefix`/`SetOption100` seems to avoid the anomaly of the "ZbInfo" key being used more than once.

**Reply by baszied** [reply]
Thanks sfronis!
The SetOption100 did the trick for me to at least get a jq parsable result

One can debate whether or not to use asynchronous MQTT or synchronous HTTP commands but if the API supports JSON responses it should imo at least be correctly interpretable..  I find that when sending the command using MQTT, the ZbInfo command will generate multiple MQTT messages, one for every device;  Using HTTP, I can get all devices info with a single synchronous command.  jq helps to filter and seach for specific devices.  In my case, I can for example run a ZbInfo and for example immediately filter all devices starting with "Smoke*" to get a list of my smoke alarms.
